src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -3095,11 +3112,9 @@ internal static Attribute GetCustomAttributeHelper(Type type, Type attributeType
 
                     if (attr != null)
                     {
-                        Type t = attr.GetType();
-
                         foreach (CustomAttributeNamedArgument namedArgument in data.NamedArguments)
                         {
-                            PropertyInfo p = t.GetProperty(namedArgument.MemberInfo.Name, BindingFlags.Public | BindingFlags.Instance)!;
+                            PropertyInfo p = attributeType.GetProperty(namedArgument.MemberInfo.Name, BindingFlags.Public | BindingFlags.Instance)!;
```

Comment:
I think this can potentially break the existing code. The attribute match above is name based (`AttributeTypeNamesMatch`) so there's no guarantee that `attributeType` and `attr.GetType()` will be type-system-related. They can be two separate types which happen to have the same name. After this change the `p.SetValue` below may fail if the types were different.

Reply:
The `attr` variable is the result of `Activator.CreateInstance(attributeType)` a couple lines above. I have a hard time coming up with a scenario where `attr.GetType() != attributeType`.

Reply:
Sorry - I haven't seen that - you're absolutely right.